csrc/host_ir/executor.cpp

```diff
@@ -46,16 +45,9 @@ HostIrExecutor::HostIrExecutor(
 bool HostIrExecutor::supported(Fusion* fusion) {
   FUSER_PERF_SCOPE("HostIrExecutor::supported");
   std::vector<Expr*> exprs = fusion->exprs();
-  if (std::any_of(exprs.begin(), exprs.end(), [](Expr* e) {
-        return isResharding(e) && HostIrLower::canLower(e);
-      })) {
+  if (std::any_of(exprs.begin(), exprs.end(), isResharding)) {
```

Comment:
`canLower` can be useful to check we do not have any non-decomposed ops in the fusion.
Alternatively, it can be added at the end of `InsertReshardingsPass`.

Reply:
> can be useful

Yes but that defeats the purpose to avoid the dependency on host_ir/lower from the FusionExecutorCache stack. Instead, we can call getCommunicationInfo at the end of InsertReshardingsPass.

Reply:
I'm hitting the expected error that getCommunicationInfo doesn't support broadcast and sendrecv. I'll try to fix that...